grunttasks/scan-email-translations.js

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (grunt) {
+  grunt.registerTask('scan-email-translations', 'Scan the generated emails for invalid style tags', [
+    'generate-valid-styles',
+    'po2json',
+    'verify-styles'
+    ]
+  );
+
+  grunt.registerMultiTask('verify-styles', 'Verify whether the translated styles are valid', function (){
+    var fileArray = this.files[0].src;
+    var contents = grunt.file.read(fileArray.shift());
+    var styleArray = JSON.parse(contents);
```

Comment:
Wildly untested counter-proposal. Instead of assuming the first item in the `src` array is your `validStyles` file/array, just pass the valid styles in as an option:

```js
  grunt.config('verify-styles', {
    dist: {
      options: {
        validStyles: grunt.file.readJSON('.tmp/validStyles.json')
      },
      files: [
        {
          expand: false,
          src: [
            '.tmp/l10n/*/messages.json'
          ]
        }
      ]
    }

  ...

  grunt.registerMultiTask('verify-styles', 'Verify whether the translated styles are valid', function (){
    var fileArray = this.files[0].src;
    var options = this.options({
      validStyles: []
    });
    var styleArray = options.validStyles;

    var hasErrors = 0;
    var styleRegex = /style="(.*?)"/ig;
```